tests/gateway/providers/test_gemini.py

```diff
@@ -412,3 +387,149 @@ async def test_invalid_parameters_chat(override, exclude_keys, expected_msg):
 
     with pytest.raises(AIGatewayException, match=expected_msg):
         await provider.chat(chat.RequestPayload(**payload))
+
+
+def chat_stream_response():
+    return [
+        b'data: {"candidates":[{"content":{"parts":[{"text":"a"}]},"finishReason":null}],"'
+        b'id":"test-id","object":"chat.completion.chunk","created":1,"model":"test"}\n',
+        b"\n",
+        b'data: {"candidates":[{"content":{"parts":[{"text":"b"}]},"finishReason":"stop"}],"'
+        b'id":"test-id","object":"chat.completion.chunk","created":1,"model":"test"}\n',
+        b"\n",
+        b"data: [DONE]\n",
+    ]
+
+
+def chat_stream_response_incomplete():
+    return [
+        b'data: {"candidates":[{"content":{"parts":[{"text":"a"}]},"finishReason":null}],"'
+        b'id":"test-id","object":"chat.completion.chunk",',
+        b'"created":1,"model":"test"}\n\n'
+        b'data: {"candidates":[{"content":{"parts":[{"text":"b"}]},"finishReason":"stop"}],"'
+        b'id":"test-id","object":"chat.completion.chunk","created":1,"model":"test"}\n',
+        b"\n",
+        b"data: [DONE]\n",
+    ]
+
+
+@pytest.mark.parametrize("resp", [chat_stream_response(), chat_stream_response_incomplete()])
+@pytest.mark.asyncio
+async def test_gemini_chat_stream(resp):
+    config = chat_config()
+    mock_client = mock_http_client(MockAsyncStreamingResponse(resp))
+    with mock.patch("time.time", return_value=1):
+        with mock.patch("aiohttp.ClientSession", return_value=mock_client) as mock_build_client:
+            provider = GeminiProvider(RouteConfig(**config))
+            payload = {"messages": [{"role": "user", "content": "Tell me a joke"}]}
+
+            stream = provider.chat_stream(chat.RequestPayload(**payload))
+            chunks = [jsonable_encoder(chunk) async for chunk in stream]
```

Comment:
Let's minimize the scope of mock context. I think only these lines need mocking.